### ISSUE TITLE
Backward Gradient calculation slow

### ISSUE THREAD

**Issue by DLExpert** [UKPLab/sentence-transformers#1009]
Hi Team,

While training new networks, we observed the backward gradient calculation is 20 times slower than forward pass.
what could be the issue here ?

**Comment by nreimers** [comment]
Yes, this is how it is with neural networks, that backprop is far slower than the forward pass.
